xdsl/dialects/scf.py

```diff
@@ -369,6 +362,64 @@ def parse(cls, parser: Parser) -> Self:
 
         return for_op
 
+    @abstractmethod
+    def _print_bounds(self, printer: Printer) -> None:
+        raise NotImplementedError()
+
+    @classmethod
+    @abstractmethod
+    def _parse_bounds(cls, parser: Parser) -> tuple[SSAValue, SSAValue]:
+        raise NotImplementedError()
+
+
+@irdl_op_definition
+class ForOp(ForRofOperation):
+    name = "scf.for"
+
+    def _print_bounds(self, printer: Printer):
+        printer.print_ssa_value(self.lb)
+        printer.print_string(" to ")
+        printer.print_ssa_value(self.ub)
+
+    @classmethod
+    def _parse_bounds(cls, parser: Parser) -> tuple[SSAValue, SSAValue]:
+        lb = parser.parse_operand()
+        parser.parse_characters("to")
+        ub = parser.parse_operand()
+        return lb, ub
+
+
+@irdl_op_definition
+class RofOp(ForRofOperation):
```

Comment:
The semantics of `scf.for` is always counting up, it cannot model `range(0, 200, -2)`. We need to count down to 0 in riscv, because comparison with 0 needs one fewer register, and we do not have the infrastructure to determine that we can reverse the iteration order, unlike LLVM. TBH we also don't want it, we might as well model the target iteration directly.

Reply:
I agree with your concern wrt mirrored dialects diverging, I don't know what to do about it. Maybe we could start having ops like `scf.xdsl.rof`

Reply:
negative steps are disallowed by `scf.for`:

> The step is a value of same type but required to be positive.

See [here](https://mlir.llvm.org/docs/Dialects/SCFDialect/#scffor-scfforop)

Reply:
Okay, I see that we should use a new operation then :+1: 
I think naming it scf.xdsl.rof or xdsl.scf.for would be the way to go.